### PR TITLE
gui: Avoid code generating HTML

### PR DESCRIPTION
**Pull request by calmh**
This changes the rendering of shared folders and devices to not use code generated HTML and tooltips. Instead we show the full list of devices/folders (allowing line breaks) and add an asterisk + tooltip to each device/folder that has an annotation.

Screenshot 2023-06-04 at 10 02 21

Screenshot 2023-06-04 at 10 02 34

Also removes the auto-HTML-ification of warning messages.

Both of these changes are necessary because the strings displayed here are unsafe: potentially controlled by someone on a remote device.

### CODE REVIEW

**Comment by acolomb** [comment]
Looks nice overall. But please don't simply use an asterisk. The superscript 1 was chosen for a good reason, as there is also a number 2 for a different cause.  I'd like to keep seeing the difference on first glance, without tooltips.

**Comment by calmh** [comment]
I removed the numbers since they no longer referred to anything, but I agree that once you learn them it adds value at a glance. Re-adding.

**Comment by acolomb** [comment]
Have you checked what this looks like in the Sharing tab under device / folder edit modals?

**Comment by calmh** [comment]
Ah, indeed, there's one more thing to fix.

**Comment by tomasz1986** [comment]
Would it be possible to update the screenshots to display the final iteration?

**Comment by calmh** [comment]
I wasn't super excited about the appearance of the spacing with superscript and comma, but I don't really care that much. Sure, have commas.

Screenshot 2023-06-04 at 13 49 24

**Comment by tomasz1986** [comment]
Thank you!

I've tested the code locally and it does look nice, although the list, which was hidden previously, can now get quite lengthy with a lot of folders, e.g.

image

On the other hand, I personally do like having them visible all the time.

**Comment by calmh** [comment]
Yes, this is the tradeoff. I think this is how it should be, if we want to list the folders in the GUI. A large list is a large list, it isn't really easier to read stuffed into a tooltip, I think.

**Review by acolomb — APPROVED** [review]

File: gui/default/index.html [hunk 1]
```diff
@@ -553,9 +553,17 @@ <h4 class="panel-title">
...
+                          <span ng-repeat="device in folder.devices">
+                            <span ng-if="device.deviceID != myID" ng-switch="completion[device.deviceID][folder.id].remoteState">
+                              <span ng-switch-when="notSharing" data-original-title="{{'The remote device has not accepted sharing this folder.' | translate}}" tooltip>{{deviceName(devices[device.deviceID])}}<sup>1</sup><span ng-if="!$last">,</span></span>
+                              <span ng-switch-when="paused" data-original-title="{{'The remote device has paused this folder.' | translate}}" tooltip>{{deviceName(devices[device.deviceID])}}<sup>2</sup><span ng-if="!$last">,</span></span>
+                              <span ng-switch-default>{{deviceName(devices[device.deviceID])}}<span ng-if="!$last">,</span></span>
+                            </span>
+
+                          </span>
                         </td>
+
```
Comment (imsodin) [inline gui/default/index.html:566]: Intentional or spurious newlines?

**Review by imsodin — APPROVED** [review]
So much on me not finding any usages of html...

One potential whitespace nit.

**Comment by tomasz1986** [comment]
Just one comment. There is a problem when using super-long, non-breaking labels, e.g.

image

The label gets truncated and cannot be viewed in full at all, which wasn't the case with the tooltip. A simple/dirty fix I can think of is to add the `file-path` class to the relevant table cell. I call it "dirty" because the class name doesn't really match the content, but it does allow the long text to break.

**Comment by calmh** [comment]
fixd

**Comment by calmh** [comment]
Screenshot 2023-06-05 at 10 28 55

**Review by acolomb — APPROVED** [review]

**Pull request merged**